lib/instance.js

```diff
@@ -322,8 +322,11 @@ Instance.prototype.set = function(key, value, options) { // testhint options:non
           // If attribute is not in model definition, return
           if (!this._isAttribute(key)) {
             if (key.indexOf('.') > -1 && this.Model._isJsonAttribute(key.split('.')[0])) {
-              Dottie.set(this.dataValues, key, value);
-              this.changed(key, true);
+              var previousDottieValue = Dottie.get(this.dataValues, key);
+              if (!_.isEqual(previousDottieValue, value)) {
```

Comment:
I wanted to make sure to do a deep comparison if you pass a object as value, but maybe that is not a valid use case. I guess you could always use a multiple dot.separated key instead. If you prefer I can update the pull request to a ```!===``` comparison?